docs/lexicon.html

```diff
@@ -240,6 +240,61 @@ <h3><a name="canonicalise">canonicalise</a></h3>
       </ul>
     </p>
 
+    <h3><a name="exec">exec</a></h3>
+
+    <p><pre class="rule"><code>exec(cmd, stdout=True, stderr=False)</code></pre></p>
+
+    <p>Executes a command and returns the output as a string if the command's
+      exit code returns <code>0</code>. If <code>stdout</code>
+      and <code>stderr</code> are both set to true, the output from both stdout
+      and stderr is interleaved.</p>
+
+    <p>For example:
+      <ul>
+        <li><code>exec(cmd = ["git", "symbolic-ref", "-q", "--short", "HEAD"])</code> -> <code>master</code></li>
+        <li><code>exec(cmd = "git rev-parse --short HEAD")</code> -> <code>26ea52b</code></li>
+        <li><code>exec(cmd = ["sh", "-c", "echo stdout1; echo 1>&2 stderr; echo stdout2"], stdout = False, stderr = True)</code> -> <code>stderr</code></li>
+        <li><code>exec(cmd = ["sh", "-c", "echo stdout1; echo 1>&2 stderr; echo stdout2"], stderr = True)</code> -> <code>stdout1\nstderr\nstdout2</code></li>
+      </ul>
+    </p>
+
+    <table>
+      <thead>
+      <tr>
+	<th>Argument</th>
+	<th>Default</th>
+	<th>Type</th>
+	<th></th>
+      </tr>
+      </thead>
+      <tbody>
+
+      <tr>
+	<td>cmd</td>
+	<td></td>
+	<td>str or list</td>
+	<td>Command to be executed.  Command does not invoke the system shell
```

Comment:
Not sure if it is worth calling out specifically, but might be a good idea to add a caution that the command should be deterministic (and reproducible with the build state), otherwise bad things w.r.t. caching and incrementality will happen. Obviously this is similar to other gotchas (such as putting timestamps in files) but every little helps.

Reply:
Updated the docs to reflect this concern.